paasta_tools/vitesscluster_tools.py

```diff
@@ -1,27 +1,550 @@
+                "key": "vault-vttablet-approle-secretid",
+            }
+        },
+    },
+]
+
+# Vault auth related variables
+VTGATE_EXTRA_ENV = [
+    {
+        "name": "VAULT_ROLEID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-roleid",
+                "key": "vault-vtgate-approle-roleid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_SECRETID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-secretid",
+                "key": "vault-vtgate-approle-secretid",
+            }
+        },
+    },
+]
+
+
+# Extra Flags
+VTADMIN_EXTRA_FLAGS = {"grpc-allow-reflection": "true"}
+
+VTCTLD_EXTRA_FLAGS = {
+    "disable_active_reparents": "true",
+    "security_policy": "read-only",
+}
+
+VTTABLET_EXTRA_FLAGS = {
+    "log_err_stacks": "true",
+    "grpc_max_message_size": "134217728",
+    "init_tablet_type": "replica",
+    "queryserver-config-schema-reload-time": "1800",
+    "dba_pool_size": "4",
+    "vreplication_heartbeat_update_interval": "60",
+    "vreplication_tablet_type": "REPLICA",
+    "keep_logs": "72h",
+    "enable-lag-throttler": "true",
+    "throttle_check_as_check_self": "true",
+    "db_charset": "utf8mb4",
+    "disable_active_reparents": "true",
+}
+
+
+def get_updated_environment_variables(
+    original_list: List[Any], update_dict: Dict[str, str]
+) -> List[Dict[str, str]]:
+    for env in original_list:
+        if env["name"] in update_dict:
+            env["value"] = update_dict[env["name"]]
+            del update_dict[env["name"]]
+    original_list.extend([{"name": k, "value": v} for k, v in update_dict.items()])
+    return original_list
+
+
+def get_affinity_spec(
+    paasta_pool: str,
+) -> Dict[str, Dict[str, Dict[str, List[Dict[str, List[Dict[str, Sequence[str]]]]]]]]:
+    spec = {
+        "nodeAffinity": {
+            "requiredDuringSchedulingIgnoredDuringExecution": {
+                "nodeSelectorTerms": [
+                    {
+                        "matchExpressions": [
+                            {
+                                "key": "yelp.com/pool",
+                                "operator": "In",
+                                "values": [paasta_pool],
+                            }
+                        ]
+                    }
+                ]
+            }
+        }
+    }
+    return spec
+
+
+def get_extra_labels(paasta_pool: str, paasta_cluster: str) -> Dict[str, str]:
+    """
+    get extra labels to adhere to paasta contract
+    """
+    extra_labels = {
+        "yelp.com/owner": "dre_mysql",
+        "paasta.yelp.com/cluster": paasta_cluster,
+        "paasta.yelp.com/pool": paasta_pool,
+    }
+    return extra_labels
+
+
+def get_extra_env(paasta_cluster: str) -> List[object]:
```

Comment:
just wanted to link to some old comments i made that seem to have gotten marked as outdated (maybe from force-pushing?) about some of these get_extra_* functions: https://github.com/Yelp/paasta/pull/3889#discussion_r1633899105

Reply:
https://github.com/Yelp/paasta/pull/3889/commits/6cbc15067cdfc3754ceaf751a8cbe8d8ee749b7b